src/rules/ResourceICUPluralTranslation.js

```diff
@@ -0,0 +1,249 @@
+                    break;
+
+                case 3:
+                    result += ` {${nodes[i].value}, date, ${nodes[i].style}} `;
+                    break;
+
+                case 4:
+                    result += ` {${nodes[i].value}, time, ${nodes[i].style}} `;
+                    break;
+
+                // for select ordinals and plurals, replace them with a fixed string
+                // so that we don't match on the details of those subparts
+                case 5:
+                    result += ` {selectordinal} `;
+                    break;
+
+                case 6:
+                    result += ` {plural} `;
+                    break;
+
+                case 7:
+                    result += " # ";
+                    break;
+
+                case 8:
+                    result += `<${nodes[i].value}/>`;
+                    break;
+
+                default:
+                    result += nodes[i].value;
+                    break;
+            }
+        }
+
+        return result.replace(/\s+/g, " ").trim();
+    }
+
+    /**
+     * Traverse an array of ast nodes to find any embedded selects or plurals
+     * or tags, and then process those separately.
+     * @private
+     */
+    traverse(resource, file, source, target) {
+        let sourcePlurals = {};
+        let targetPlurals = {};
+        let sourceTags = {};
+        let targetTags = {};
+
+        // Find the plurals and tags in this string and remember them according to their
+        // unique name. We do this because the order of plurals and tags may change in
+        // translations, so we have to go by the only part that doesn't change -- the name
+        for (let i = 0; i < source.length; i++) {
+            // selectordinal or plural
+            if (source[i].type === 5 || source[i].type === 6) {
+                sourcePlurals[source[i].value] = source[i];
+            } else if (source[i].type === 8) {
+                sourceTags[source[i].value] = source[i];
+            }
+            if (target[i].type === 5 || target[i].type === 6) {
+                targetPlurals[target[i].value] = target[i];
+            } else if (target[i].type === 8) {
+                targetTags[target[i].value] = target[i];
+            }
+        }
+
+        // for each plural, try to match it up with the target plural by name and check if
+        // there is a translation
+        let results = Object.keys(sourcePlurals).map(name => {
+            const sourcePlural = sourcePlurals[name];
+            const targetPlural = targetPlurals[name];
+            if (!targetPlural) {
+                // missing target plurals are for a different rule, so don't report it here
+                return;
+            }
+            return Object.keys(sourcePlural.options).map(category => {
+                const sourceStr = this.reconstruct(sourcePlural.options[category].value);
+                const targetStr = this.reconstruct(targetPlural.options[category].value);
```

Comment:
Mismatched categories should also be accounted for to avoid potential TypeError from accessing value on undefined

```suggestion
                if (!targetPlural.options[category]) { return; }
                const targetStr = this.reconstruct(targetPlural.options[category].value);
```